packages/block-editor/src/components/global-styles/background-panel.js

```diff
@@ -845,14 +1051,57 @@ export default function BackgroundPanel( {
 							inheritedValue={ resolvedInheritedValue }
 							defaultValues={ defaultValues }
 							onResetImage={ () => {
-								setIsDropDownOpen( false );
 								resetBackground();
 							} }
-							onRemoveImage={ () => setIsDropDownOpen( false ) }
 						/>
 					) }
 				</ToolsPanelItem>
-			</div>
+				{ shouldShowBackgroundColorControls &&
+					path === '/background' && (
+						<ToolsPanelItem
+							hasValue={ hasBackground }
+							label={ __( 'Color' ) }
+							onDeselect={ resetBackgroundColor }
+							isShownByDefault={ defaultControls.background }
+							panelId={ panelId }
+						>
+							<ColorPanelDropdown
+								key="background"
+								label={ __( 'Color' ) }
+								hasValue={ hasBackground }
+								resetValue={ resetBackgroundColor }
+								isShownByDefault={ defaultControls.background }
+								indicators={ [ gradient ?? backgroundColor ] }
+								tabs={ [
+									hasSolidColors && {
+										key: 'background',
+										label: __( 'Background' ),
+										inheritedValue: backgroundColor,
+										setValue: setBackgroundColor,
+										userValue: userBackgroundColor,
+									},
+									hasGradientColors && {
+										key: 'gradient',
+										label: __( 'Gradient' ),
+										inheritedValue: gradient,
+										setValue: setGradient,
+										userValue: userGradient,
+										isGradient: true,
+									},
+								].filter( Boolean ) }
```

Comment:
Potentially premature optimisation, but as of now the `tabs` array is being re-creater every time the component re-renders.